client/pk-console.c

```diff
@@ -1130,7 +1130,7 @@ pk_console_update_system (PkConsoleCtx *ctx, GError **error)
 
 	/* do the async action */
 	sack = pk_results_get_package_sack (results);
-	package_ids = pk_package_sack_get_ids (sack);
+	package_ids = pk_package_sack_get_ids_without_removals (sack);
```

Comment:
I think the filtering needs to be done here

Reply:
I've found 'pk_package_sack_remove_by_filter' function. I'll rework it using this function.